src/app/Fake.DotNet.Testing.Coverlet/Coverlet.fs

```diff
@@ -0,0 +1,119 @@
+type OutputFormat =
+    | Json
+    | Lcov
+    | OpenCover
+    | Cobertura
+    | TeamCity
+
+/// The type of coverage to use when failing under a threshold.
+type ThresholdType =
+    | Line
+    | Branch
+    | Method
+
+/// The statistic to use when failing under a threshold.
+type ThresholdStat =
+    | Minimum
+    | Total
+    | Average
+
+/// Coverlet MSBuild parameters. For more details see: https://github.com/tonerdo/coverlet/blob/master/Documentation/MSBuildIntegration.md
+type CoverletParams =
+    { /// (Required) Format of the generated output.
+        OutputFormat : OutputFormat
+        /// (Required) Path to the generated output file, or directory if it ends with a `/`.
+        Output : string
+        /// Namespaces to include, as (AssemblyName, Namespace) pairs. Supports `*` and `?` globbing.
+        Include : (string * string) list
+        /// Namespaces to exclude, as (AssemblyName, Namespace) pairs. Supports `*` and `?` globbing.
+        Exclude : (string * string) list
+        /// Exclude methods, types and assemblies annotated with these attributes.
+        ExcludeByAttribute : string list
+        /// Exclude these source files. Supports path globbing.
+        ExcludeByFile : string list
+        /// Coverlet json file to merge with the output of this run.
+        MergeWith : string option
+        /// Minimum coverage percent. Build fails if the result is below.
+        Threshold : int option
+        /// Type of coverage to check against the threshold.
+        ThresholdType : ThresholdType
+        /// Coverage statistic to check against the threshold.
+        ThresholdStat : ThresholdStat
+        /// Generate results with URL links from SourceLink instead of file paths.
+        UseSourceLink : bool }
+
+/// The default parameters.
+let Defaults =
+    { OutputFormat = OutputFormat.Json
+      Output = "./"
+      Include = []
+      Exclude = []
+      ExcludeByAttribute = []
+      ExcludeByFile = []
+      MergeWith = None
+      Threshold = None
+      ThresholdType = ThresholdType.Line
+      ThresholdStat = ThresholdStat.Minimum
+      UseSourceLink = false }
+
+let private outputFormatToString = function
+    | OutputFormat.Json -> "json"
+    | OutputFormat.Lcov -> "lcov"
+    | OutputFormat.OpenCover -> "opencover"
+    | OutputFormat.Cobertura -> "cobertura"
+    | OutputFormat.TeamCity -> "teamcity"
+
+let private namespacesToString =
+    Seq.map (fun (asm, ns) -> "[" + asm + "]" + ns)
+    >> String.concat ","
+
+let private thresholdTypeToString = function
+    | ThresholdType.Line -> "line"
+    | ThresholdType.Branch -> "branch"
+    | ThresholdType.Method -> "method"
+
+let private thresholdStatToString = function
+    | ThresholdStat.Minimum -> "minimum"
+    | ThresholdStat.Total -> "total"
+    | ThresholdStat.Average -> "average"
+
+/// Add Coverlet parameters to the MSBuild command.
+let withMSBuildArguments (param: CoverletParams -> CoverletParams) (args: MSBuild.CliArguments) =
```

Comment:
Interesting approach, currently we don't have any module that works like this. Is there any other API design you have considered (I mean it definitely looks reasonable to me, just asking)

In any case, I think we need inline docs for this as well.